### ISSUE TITLE
Call osm-search from lisp

### ISSUE THREAD

**Issue by hmelman** [minad/osm#14]
I'd like to be able to do something like the following: `(osm-search "123 Main St., Springfield")`

My use case is being in another buffer with an address and wanting to see it on a map.  I only have an address, not lat/lon.

**Comment by minad** [comment]
Yes would be nice to have. The search functionality needs more love anyway.

1. Support for multiple geoname services, not only nominatim
2. Support for querying navigation/routing services

**Comment by minad** [comment]
I just looked again at the code. The reason why I didn't implement lisp arguments to `osm-search` was that the process is two step. First we search, then we select from the candidates. For the lisp interface we could just take the first candidate always or open the completion interface in all cases?

**Comment by hmelman** [comment]
In my use case I'm fine opening the completion interface (I probably won't have a complete address anyway).  If there's only one candidate you could skip the selection step.  I suppose some callers will want each method, though since you're popping open a map, it seems like there will always be an interactive user so selection is fine. ¯\_(ツ)_/¯

**Comment by hmelman** [comment]
Thanks, that works great.  Question about the transient pin that is put after the search.  1. I like it.  2. The first thing I did was zoom out and it disappeared, is it supposed too?  Is there a way to make it remain?  

Okay, 3. I see I can set a transient pin by `osm-center-click`, is there a way to do it via the keyboard, say at the center of the screen?  Maybe that doesn't make sense at all, but after I zoomed out and it disappeared, I wanted a way to mark the center of the screen as I zoomed to help me track the position for further zooms.  And I didn't want something as permanent as a bookmark and clicking at that point would be imprecise.
